### PR TITLE
Initial Pull Request for Adding the Oniro IDE Documentation

### PR DESCRIPTION
**Pull request by marzi333**
This is the first pull request to add the documentation for the [Oniro IDE](https://github.com/eclipse-oniro4openharmony/oniro-ide).
This commit includes the user interface guide for the IDE.

### CODE REVIEW

**Comment by frankplus** [comment]
looks good as an initial PR, can we add instructions on how to install the IDE?

**Comment by Stefan-Schmidt** [comment]
> looks good as an initial PR, can we add instructions on how to install the IDE?

This got added now.

**Review by jarmark — CHANGES_REQUESTED** [review]
A set of minor style improvements. Nice work overall.

**Comment by Stefan-Schmidt** [comment]
When all review comments are fixed it would also be good to squash this into one commit that is ready to be merged.

File: ide-getting-started.md [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+title: Getting Started
+layout: home
+nav_order: 1
+parent: Oniro IDE
+---
+## Installation
+Please make sure that your system follows the [Theia development prerequisites](https://github.com/eclipse-theia/theia/blob/master/doc/Developing.md#prerequisites).
+
+### Building
+You can build both browser and electron application by running the following script:
```
[the remark this answers is not in the document]
Comment (marzi333) [reply ide-getting-started.md:12]: Specified: The IDE can be built & run in your browser, or as a packaged desktop application using [Electron](https://www.electronjs.org/).

File: ide-getting-started.md [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+```
+yarn browser start
+```
+And similarly for the electron application:
+
+```
+yarn electron start
+```
+
+Watching
```
[the remark this answers is not in the document]
Comment (marzi333) [reply ide-getting-started.md]: Should've been a subtitle.

File: ide-code-editing.md [hunk 3]
```diff
@@ -0,0 +1,29 @@
+---
+title: Code Editing
+layout: home
+nav_order: 3
+parent: Oniro IDE
+---
+
+## Keyboard Shortcuts
+Being able to keep your hands on the keyboard when writing code is crucial for high productivity. You can explore the different keyboard shortcuts by navigating to `File -> Preferences -> Keyboard Shortcuts` from the menu tab or clicking `Ctrl+Alt+,`.
```
[the remark this answers is not in the document]
Comment (marzi333) [reply ide-code-editing.md]: changed to various.

File: ide-code-editing.md [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+
+## Selection
+Different selection options can be explored from `Selection` in the menu tab. A few of the important ones are `Shift + Right/Left/Up/Down` for basic selection, `Ctrl + Shift + Right/Left`  for whole words selection, and column selection by placing the cursor in one corner and then holding `Shift+Alt` while dragging to the opposite corner.
+
+## Save & Auto Save
+To save a file press `Ctrl+S` or enable auto save from `File -> Auto Save` which will save your changes after a configured delay or when focus leaves the editor.
+
+## Find & Replace
+![Alt text](assets/images/find-and-replace.png)
+Press `Ctrl+F` to toggle the `Find` feature of the IDE, write the word or phrase you're searching for and press `Enter` or `Shift+Enter` to go up and down the results. As seen in the above there are several options to the search namely:
```
[the remark this answers is not in the document]
Comment (marzi333) [reply ide-code-editing.md]: forgot to add "figure", changed to: "As seen in the above figure, there..."

File: ide-code-navigation.md [hunk 5]
```diff
@@ -0,0 +1,34 @@
+---
+title: Code Navigation
+layout: home
+nav_order: 4
+parent: Oniro IDE
+---
+
+A dynamic combination between programming languages services and quick text editing gives Oniro IDE its power. In this topic we will give an overview of a few code navigation features which can be explored when you right click on a symbol (variable, function, method, class, etc.).
```
[the remark this answers is not in the document]
Comment (marzi333) [reply ide-code-navigation.md]: changed to the above.

File: ide-code-editing.md [hunk 6]
```diff
@@ -0,0 +1,29 @@
...
+
+1. **Match Case:** to find a word or phrase with the same case letters.
+2. **Match Whole Word:** to eliminate words or phrases that don't contain all the letters of each word from the results.
+3. **Use Regular Expression:** For an advanced search using [regular expresions](https://learn.microsoft.com/en-us/visualstudio/ide/using-regular-expressions-in-visual-studio?view=vs-2022).
+4. **Find in Selection:** to first select a piece of code and then search only within the selection.
+5. **Replace:** to replace each of the search results with the word or phrase in the *Replace* field.
```
[the remark this answers is not in the document]
Comment (marzi333) [reply ide-code-editing.md]: yes, that's better phrasing.

File: ide-getting-started.md [hunk 7]
```diff
@@ -0,0 +1,65 @@
...
+### Building
+You can build both browser and electron application by running the following script:
+```
+yarn
+yarn compile
+yarn download:plugins
+```
+
+### Running
+To start the browser application you just need to run:
```
[the remark this answers is not in the document]
Comment (marzi333) [reply ide-getting-started.md]: removed it.

File: oniro-ide.md [hunk 8]
```diff
@@ -0,0 +1,21 @@
...
+2. User Interface (ide-user-interface)
+3. Code Editing (ide-code-editing)
+4. Code Navigation (ide-code-navigation) 
+5. Refactoring (ide-refactoring)
+6. Building and Debugging (ide-building-and-debugging)
+7. Language Support (ide-language-support)
+-->
+
+# Oniro IDE
+Oniro IDE is an [Eclipse Theia](https://theia-ide.org/) based IDE built on the Oniro Operating System. It allows developers to build applications on a wide range of smart devices in what is called "One-time development". One set of code can adapt to multiple ends improving development efficiency and implementing cross-end service ecosystem sharing.
```
[the remark this answers is not in the document]
Comment (marzi333) [reply oniro-ide.md]: Changed. Thank you for correcting this.

File: ide-user-interface.md [hunk 9]
```diff
@@ -0,0 +1,49 @@
...
+
+![](assets/images/Main%20Layout.png)
+
+## Basic Layout
+The basic layout consists of 8 main areas:
+1. **Menu Bar**: that contains the main menus such as File, Edit, View, Run..etc.
+1. **Tool Bar**: for going home, and toggling the explorer and debugging side bars.
+1. **Layout Bar**: for changing the layout and positioning of the tool bar.
+1. **Editor Groups**: with all the different files you are editing. You open as many files as you want and stack them horizontally and vertically.
+1. **Primary Side Bar**: usually hosts the `Explorer` view to assist you to navigate and work on your project.
```
[the remark this answers is not in the document]
Comment (marzi333) [reply ide-user-interface.md]: Fixed.

File: ide-user-interface.md [hunk 10]
```diff
@@ -0,0 +1,49 @@
...
+* Create, delete, rename files and folders by right click.
+* Move files and folders by drag and drop.
+* Duplicate, copy and paste files into a folder.
+
+{: .note }
+> You could run commands in a folder by right clicking on the folder itself or a file in the folder and choosing `Open in Terminal` from the pop out menu.
+
+
+### Open Editors
+On the top of the `Explorer` side view you can find a pane called `Open Editors` this shows all the files that you currently have open. This allows for quick navigation between the opened editor tabs.
```
[the remark this answers is not in the document]
Comment (marzi333) [reply ide-user-interface.md]: Fixed.

File: ide-user-interface.md [hunk 11]
```diff
@@ -0,0 +1,49 @@
...
+
+## Views
+
+The `Explorer` is one of multiple other views available in Oniro IDE. 
+![](assets/images/view-menu.png)
+By navigating to `View` from the menu bar you could see the different views that can be opened such as:
```
[the remark this answers is not in the document]
Comment (marzi333) [reply ide-user-interface.md]: Corrected. Thanks!

**Review by Stefan-Schmidt — APPROVED** [review]

**Review by frankplus — APPROVED** [review]
LGTM

**Pull request merged**